source/braille.py

```diff
@@ -1000,8 +987,27 @@ def update(self):
 		self.selectionStart = self.selectionEnd = None
 		self._isFormatFieldAtStart = True
 		self._skipFieldsNotAtStartOfNode = False
-
 		self._endsWithField = False
+		# Get the selection.
+		sel = self._getSelection()
+		if not sel.isCollapsed:
+			self._readingInfo = readingInfo = sel.copy()
+			if self.obj.isSelectionAnchoredAtStart:
+				# The end of the range is exclusive, so make it inclusive first.
+				readingInfo.move(textInfos.UNIT_CHARACTER, -1, "end")

```

Comment:
I'm still not happy about this. For a downward multi-line selection it is more practical to see the next line to be selected rather than the very right edge of the last selected line. But there is no good way to detect the difference between the result of pressing shift+downArrow and shift+end. The only solution I know of is to rely on cursor position even when there is a selection, but that doesn't work in all applications.

So there are two options:
1. Rely on the cursor where possible. Provides great usability in some programs (Notepad++, Eclipse) and average usability in others (Notepad, Word, browse mode).
2. Don't try to be clever and always show the very edge of a selection. This is consistent and predictable, but usability is so-so.